api/v1alpha4/machinedeployment_webhook.go

```diff
@@ -92,6 +93,34 @@ func (m *MachineDeployment) validate(old *MachineDeployment) error {
 		)
 	}
 
+	if m.Spec.Strategy != nil &&
+		m.Spec.Strategy.RollingUpdate != nil {
+		total := 1
+		if m.Spec.Replicas != nil {
+			total = int(*m.Spec.Replicas)
+		}
+
+		if m.Spec.Strategy.RollingUpdate.MaxSurge != nil {
+			if _, err := intstrutil.GetScaledValueFromIntOrPercent(m.Spec.Strategy.RollingUpdate.MaxSurge, total, true); err != nil {
+				allErrs = append(
+					allErrs,
+					field.Invalid(field.NewPath("spec", "strategy", "rollingUpdate", "maxSurge"),
+						m.Spec.Strategy.RollingUpdate.MaxSurge, fmt.Sprintf("invalid value: %v", err.Error())),
```

Comment:
Looking at the *current* implementation of `GetScaledValueFromIntOrPercent` an error would read like: `invalid value: invalid value for IntOrString..`

Not sure what's more important, not relying on the exact error produced by `GetScaledValueFromIntOrPercent` or having an error text which is a bit less redundant

Reply:
make sense, changed and aligned with the one in mhc.